main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -895,6 +895,71 @@ object Weeder {
           }
       }
 
+    case ParsedAst.Expression.ForEachYield(sp1, frags, exp, sp2) => {
+      //
+      // Rewrites a foreach-yield loop into Collectable.collect
+      //
+      val baseLoc = mkSL(sp1, sp2).asSynthetic
+
+      // Declare functions
+      val fqnEmpty = "Iterator.empty"
+      val fqnSingleton = "Iterator.singleton"
+      val fqnFlatMap = "Iterator.flatMap"
+      val fqnIterator = "Iterable.iterator"
+      val fqnCollect = "Collectable.collect"
+
+      // Make region variable
+      val regionSym = Name.Ident(sp1, "ForEachYieldIterRegion" + flix.genSym.freshId(), sp2)
```

Comment:
Should be lowercase. Let's also add a `Flix.Delimiter` in the middle so it's clear that it is internal.

Reply:
Also I think there's something like `Name.SyntheticIdent`?